File: Python/numberfun.py
'''
 *
 *	Problem: "Number Fun"
 *
'''

import sys
import math

cases = input()
cases = int(cases)

index = 1
while( index <= cases):

	line = input().split()
	a = float( line[0] )
	b = float( line[1] )
	c = float( line[2] )
	
	if ( a + b == c ):
		print("Possible")
	elif ( a * b == c ):
		print("Possible")
	elif ( a - b == c ):
		print("Possible")
	elif ( b - a == c ):
		print("Possible")
	elif ( (a / b == c) and (a % b == 0) ):
		print("Possible")
	elif ( (b / a == c) and (b % a == 0) ):
		print("Possible")
	else:
		print("Impossible")
	
	index += 1

File: Ruby/numberfun.rb
=begin
 *
 *	Problem:		Number Fun
 *	Problem ID:	numberfun
 *	url:				https://open.kattis.com/problems/numberfun
 *	Solution:		Ruby
 *
=end

class Solver

	def input_to_float(input)
		input_array = input.chomp.split(" ")
		input_float = input_array.map(&:to_f)
		return input_float
	end
	
	def solve
		number_cases = gets.chomp.to_i
		for this_case in 1..number_cases
			a, b, c = input_to_float(gets)
			
			if (a + b == c)
				puts "Possible"
			elsif (a * b == c)
				puts "Possible"	
			elsif (a - b == c)
				puts "Possible"
			elsif (b - a == c)
				puts "Possible"
			elsif ( (a / b == c) and (a % b == 0) )
				puts "Possible"
			elsif ( (b / a == c) and (b % a == 0) )
				puts "Possible"
			else
				puts "Impossible"
			end
		end
		
	end
	
end

solver = Solver.new
solver.solve
